File: practice/tag-service/src/main/java/com/tagservice/domain/TagService.java
package com.tagservice.domain;

import com.tagservice.domain.command.TagCommand;
import com.tagservice.infrastructure.TagRepository;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;
import reactor.core.publisher.Flux;
import reactor.core.publisher.Mono;

import java.util.List;

@Service
@RequiredArgsConstructor
public class TagService {
    private final TagRepository tagRepository;

    public Mono<Tag> registerTag(TagCommand tagCommand) {
        Tag tag = tagCommand.toEntity();
        return tagRepository.save(tag);
    }

    public Flux<Tag> getTags() {
        return tagRepository.findAll();
    }

    public Mono<List<Tag>> getTagsOfMember(String createdBy) {
        Flux<Tag> tags = tagRepository.findByCreatedBy(createdBy);

        try {
            Thread.sleep(2000);
        } catch (InterruptedException e) {
            throw new RuntimeException(e);
        }

        return tags.collectList();
    }
}

File: member-service/src/main/java/com/memberservice/domain/TagService.kt
package com.memberservice.domain

import lombok.RequiredArgsConstructor
import org.springframework.stereotype.Service
import org.springframework.web.reactive.function.client.WebClient
import reactor.core.publisher.Mono


@Service
@RequiredArgsConstructor
class TagService(
    private val webClient: WebClient
) {
    fun getTagsOfMember(memberId: String): List<String> {
        val tagList: Mono<List<String>> = webClient
            .get()
            .uri("http://localhost:7070/tag/{memberId}", memberId)
            .retrieve()
            .bodyToFlux(String::class.java)
            .collectList()

        return tagList.block() ?: emptyList()
    }
}